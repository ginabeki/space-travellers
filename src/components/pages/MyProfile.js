import React from 'react';

const MyProfile = () => (
  <div>MyProfile</div>
);

export default MyProfile;
