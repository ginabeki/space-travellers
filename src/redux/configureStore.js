import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { missionsReducer } from './missions/missions';
import { rocketsReducer } from './rockets/rockets';

const rootReducer = combineReducers({
  missions: missionsReducer,
  rockets: rocketsReducer,
});

export const store = configureStore({
  reducer: rootReducer,
});

export default { store };
