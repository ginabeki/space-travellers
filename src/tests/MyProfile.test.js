import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../redux/configureStore';
import { MyProfile } from '../components/pages/MyProfile';

describe('MyProfile Component Test ', () => {
  it('Expected to render MyProfile contents', () => {
    const ProfileComponent = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <MyProfile />
        </Router>
      </Provider>,
    ).toJSON();
    expect(ProfileComponent).toMatchSnapshot();
  });
});
