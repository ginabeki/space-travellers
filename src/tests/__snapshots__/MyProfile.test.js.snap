// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile Component Test  Expected to render MyProfile contents 1`] = `
<div
  className="d-flex w-75 mx-auto mt-4"
>
  <div
    className="my-missions w-50 me-2"
  >
    <h4>
      Rockets
    </h4>
    <ul
      className="list-group"
    />
    <h3
      className="text-danger"
    >
      No rockets reserved yet
    </h3>
  </div>
  <div
    className="my-missions w-50 me-2"
  >
    <h4>
      My Missions
    </h4>
    <ul
      className="list-group"
    />
    <h3
      className="text-danger"
    >
      You have not joined any mission yet
    </h3>
  </div>
</div>
`;
